asistente_ladm_col/gui/toolbar.py

```diff
@@ -113,15 +113,36 @@ def build_boundary(self, db):
                 selected_boundaries_layer = processing.run("native:saveselectedfeatures", {'INPUT': layers[db.names.LC_BOUNDARY_T], 'OUTPUT': 'TEMPORARY_OUTPUT'})['OUTPUT']
                 build_boundaries_layer = self.geometry.build_boundaries(selected_boundaries_layer)
 
-                with edit(layers[db.names.LC_BOUNDARY_T]):
-                    # Delete selected features as they will be imported again from a newly created layer after processed
-                    layers[db.names.LC_BOUNDARY_T].deleteSelectedFeatures()
+                copy_boundary_layer = self.app.core.get_layer_copy(layers[db.names.LC_BOUNDARY_T])
+                boundaries_count = layers[db.names.LC_BOUNDARY_T].featureCount()
+                selected_boundaries_count = layers[db.names.LC_BOUNDARY_T].selectedFeatureCount()
+                build_boundaries_count = build_boundaries_layer.featureCount()
+                expected_boundaries_count = boundaries_count - selected_boundaries_count + build_boundaries_count
 
-                # Bring back the features we deleted before, but this time, with the boundaries fixed
-                self.app.core.run_etl_model_in_backgroud_mode(db, build_boundaries_layer, db.names.LC_BOUNDARY_T, False)
+                # Build boundaries should have generated at least one boundary.
+                if selected_boundaries_count > 0 and build_boundaries_count > 0:
+                    with edit(layers[db.names.LC_BOUNDARY_T]):
+                        # Delete selected features as they will be imported again from a newly created layer after processed
+                        layers[db.names.LC_BOUNDARY_T].deleteSelectedFeatures()
+
+                    # Bring back the features we deleted before, but this time, with the boundaries fixed
+                    self.app.core.run_etl_model_in_backgroud_mode(db, build_boundaries_layer, db.names.LC_BOUNDARY_T, False)
+
+                    # check if features were insert successful
+                    if layers[db.names.LC_BOUNDARY_T].featureCount() == expected_boundaries_count:
+                        self.logger.info_msg(__name__, QCoreApplication.translate("ToolBar",
+                                                                                  "{} feature(s) was(were) analyzed generating {} boundary(ies)!").format(num_boundaries, build_boundaries_layer.featureCount()))
+
+                    else:
+                        # Clean layer because wrong data could be insert previously
```

Comment:
could have been inserted previously

Reply:
Adjustment made, available in commit 513f31a